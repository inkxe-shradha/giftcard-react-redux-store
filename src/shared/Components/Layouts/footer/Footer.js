import { Link } from "react-router-dom";
import { Typography } from "@mui/material";
import React from "react";

const Footer = () => {
  return (
    <footer className="mt-3">
      <Typography variant="body2" color="textSecondary" align="center">
        {"Made with ❤️ by Shrdha. Copyright © "}
        <Link color="inherit" to="/">
          Your Website
        </Link>{" "}
        {new Date().getFullYear()}
        {"."}
      </Typography>
    </footer>
  );
};

export default Footer;
