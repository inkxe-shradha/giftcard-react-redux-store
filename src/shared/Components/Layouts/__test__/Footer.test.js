import { render, screen } from "@testing-library/react";
import { shallow } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import Footer from "../footer/Footer";

describe("Footer test suite", () => {
  it("Should create a snapshot of the footer  page", () => {
    const { asFragment } = render(
      <Router>
        <Footer />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should get the current year in the document page", () => {
    const warper = shallow(<Footer />);

    const currentYear = new Date().getFullYear();
    expect(warper.text().includes(currentYear)).toBe(true);
  });
});
