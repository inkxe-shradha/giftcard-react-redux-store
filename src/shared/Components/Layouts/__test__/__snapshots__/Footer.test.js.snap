// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer test suite Should create a snapshot of the footer  page 1`] = `
<DocumentFragment>
  <footer
    class="mt-3"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
    >
      Made with ❤️ by Shrdha. Copyright © 
      <a
        color="inherit"
        href="/"
      >
        Your Website
      </a>
       2022.
    </p>
  </footer>
</DocumentFragment>
`;
